docs/development.md

```diff
@@ -42,6 +52,82 @@ Use Ctrl-C to stop
 
 Default username and password for admin are: `admin@example.com` and `test123`.
 
+### Running the extension as an engine in a Rails app
```

Comment:
I think I don't fully understand why we need to provide two different development options. Shouldn't it be enough to have the sandbox load `solidus_starter_frontend` as a gem? If we point to the local gem (one directory up), changes will also be reflected. WDYT?

Reply:
@waiting-for-dev The two development options have different goals:

1. As a gem contributor, I want a easy way to test if a `solidus_starter_frontend` can generate a working frontend for a Rails app. 
2. As a gem contributor, I want a easy way to preview changes I make to the app code of `solidus_starter_frontend`.

Previously, the sandbox script was both loading the gem _and_ generating the frontend. In #166, I changed the sandbox script to fulfill _only_ goal 1, based on the reasoning that the gem will be used only as a generator in actual usage.

Goal 2, on the other hand, is more of a nice-to-have for the gem contributor, since it just makes it easier for the contributor to preview changes they make to the app code.

We could change the sandbox script to load the gem instead of using it to generate the frontend. Perhaps I might be misunderstanding what a sandbox app is supposed to do, based on conventions we follow with other Solidus extensions?

Your thoughts?

Reply:
Hmmm yeah, it makes sense. Ideally, I'd like to confirm that the `solidus_starter_frontend` executable can generate a working frontend in the integrated test suite. Still, if that's not possible, then I agree we need some way of at least doing it manually.

> Perhaps I might be misunderstanding what a sandbox app is supposed to do, based on conventions we follow with other Solidus extensions?

What do you mean by the conventions we follow with other extensions? I haven't worked with Solidus extensions, so I'm unfamiliar. Isn't the sandbox app there picking changes made to the extension code? At least, it's what I do on the main Solidus repo, so I had the idea it was one of its purposes.

Anyway, yeah, probably we need both strategies, but I definitely would automate both of them. Maybe we could even provide a flag to `bin/sandbox` so switch from one to the other, defaulting to the as-gem version as I think it'll be more common. However, if we take this direction, it'd be nice to be able to configure the application name so that generating it with another strategy doesn't forcefully means overwriting the previous one (we could even use always two different names, like `sandbox_as_gem` and `sandbox_as_exec` :upside_down_face: )

I'd like to ping @kennyadsl to hear what are his thoughts on this, too.

Reply:
@waiting-for-dev I agree with defaulting to as-gem version since 1) this seems to be the more common way you'd want to use the sandbox, and 2) based on what you said, it's how the sandbox behaves in the main Solidus repo. 

I also like the idea of have two different scripts/apps for `sandbox_as_gem` and `sandbox_as_exec`.